rm-datasource/src/main/java/io/seata/rm/datasource/ConnectionContext.java

```diff
@@ -257,6 +267,24 @@ public void setAutoCommitChanged(boolean autoCommitChanged) {
         this.autoCommitChanged = autoCommitChanged;
     }
 
+    /**
+     * Gets applicationData.
+     *
+     * @return the application data
+     */
+    public String getApplicationData() throws TransactionException {
+        boolean autoCommit = this.isAutoCommitChanged();
+        // when transaction are enabled, it must be false
+        if (!autoCommit) {
+            this.applicationData.put(AUTO_COMMIT, autoCommit);
+            try {
+                return MAPPER.writeValueAsString(this.applicationData);
```

Comment:
Is there a JDK implementation class for converting to JSON that doesn't depend on Jackson?